### PR TITLE
Update HIP-583 design doc and test plan

### LINKED ISSUE #4390: Update design doc and test plan for HIP-583
### Problem

Since the initial design doc for HIP-583 was created there have been a few updates to the HIP most notably https://github.com/hashgraph/hedera-improvement-proposal/pull/673 
We want to include all finalized changes in the design doc itself and to update the test plan spec accordingly.

### Solution

Update https://github.com/hashgraph/hedera-services/blob/develop/docs/design/alias-configuration.md

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by dimitar-dinev**
**Description**:

Modifies the `alias-configuration.md` HIP-583 design doc with the final architecture, logic, and behavior.
Also updates the HIP-583 test plan document.

**Related issue(s)**:

Fixes #4390

### CODE REVIEW

**Review by stoqnkpL — COMMENTED** [review]
Not sure how this document ended up in the `token-service/` folder but I don't think this is the right place for it. Can we move it to a new `crypto-service/` or `account-service/` folder?

**Comment by dimitar-dinev** [comment]
> Not sure how this document ended up in the `token-service/` folder but I don't think this is the right place for it. Can we move it to a new `crypto-service/` or `account-service/` folder?

AFAIK, with the modularization effort [the `CryptoService` and `TokenService` are both part of the `token-service`](https://github.com/hashgraph/hedera-services/tree/develop/hedera-node/hedera-token-service/src/main/java/com/hedera/node/app/service/token), so maybe that explains the new path 🤔

**Review by stoqnkpL — APPROVED** [review]
LGTM! Very well written and detailed!!

**Comment by david-bakin-sl** [comment]
Reviewing this (complex&complete!) doc - but one note right away:  This doc is rather schizophrenic referring sometimes to lazy accounts, sometimes to hollow accounts, and sometimes to both as "hollow/lazy"!  I think readability could be confused by choosing and using _one or the other_ throughout and then add a prominent note at the top: "hollow is a synonym for lazy" or something like that!

**Comment by dimitar-dinev** [comment]
> I'm reviewing this (complex&complete!) doc, in progress - but one note right away: This doc is rather schizophrenic referring sometimes to lazy accounts, sometimes to hollow accounts, and sometimes to both as "hollow/lazy"! I think readability could be improved by choosing and using _one or the other_ throughout and then add a prominent note at the top: "hollow is a synonym for lazy" or something like that!

Agreed, this may be confusing, especially for a new reader --- [changed the wording and explanations with this commit](https://github.com/hashgraph/hedera-services/pull/4818/commits/22897449f5a92e9b67f5590eb145561be8a7d90c), does it read easier and clearer?

**Comment by david-bakin-sl** [comment]
> Agreed, this may be confusing, especially for a new reader --- [changed the wording and explanations with this commit](https://github.com/hashgraph/hedera-services/pull/4818/commits/22897449f5a92e9b67f5590eb145561be8a7d90c), does it read easier and clearer?

Much more clear (to me!)  Distinguishes between the "hollow account" itself and its "lazy creation".   (Optional: Do you think a specific sentence saying exactly this would help?)

File: docs/design/services/token-service/alias-configuration.md [hunk 1]
```diff
@@ -1,67 +1,148 @@
...
+As part of [HIP 32](https://github.com/hashgraph/hedera-improvement-proposal/blob/master/HIP/hip-32.md) the functionality was made available via the auto-create flow in which an `ED25519` or `ECDSA` public key was provided and used as the alias.
 
-To support greater scenario especially smart contract related scenarios for ECDSA based key the ledger needs to support additional ways of setting the alias at creation.
+To support greater scenarios, especially smart contract related scenarios for ECDSA-based key, the ledger needs to support additional ways of setting the alias at creation.
 
-This document explains how additional Crypto transaction flows may be used to create an account with an alias.
+This document explains how additional crypto and smart contract service transaction flows may be used to create an account with an alias.
+
+## Prerequisite reading
+* [HIP 32](https://github.com/hashgraph/hedera-improvement-proposal/blob/master/HIP/hip-32.md)
```
Comment (david-bakin-sl) [inline docs/design/services/token-service/alias-configuration.md:12]: Should probably refer directly to `hips.hedera.com` like the next line.  (Also, nit: `HIP-32` with dash for consistency.)

**Comment by Nana-EC** [comment]
@stoqnkpL i think this can be completed now that HIP 583 has been updated

File: docs/design/services/token-service/alias-configuration.md (hunk 1 above)
Comment (MiroslavGatsanoga) [reply docs/design/services/token-service/alias-configuration.md]: Fixed.

**Review by Nana-EC — APPROVED** [review]

**Review by tinker-michaelj — APPROVED** [review]
LGTM!

**Pull request merged**